crates/gitbutler-tauri/src/virtual_branches.rs

```diff
@@ -478,12 +478,16 @@ pub mod commands {
         projects
             .update(&projects::UpdateRequest {
                 id: project.id,
-                project_data_last_fetched: Some(project_data_last_fetched),
+                project_data_last_fetched: Some(project_data_last_fetched.clone()),
                 ..Default::default()
             })
             .await
             .context("failed to update project with last fetched timestamp")?;
 
+        if let FetchResult::Error { error, .. } = project_data_last_fetched {
```

Comment:
This allows the frontend to respond to errors during fetching, which otherwise would have been swallowed here.